### DISCUSSION TITLE
How can we best manage variant name clashes in PyPI?

### DISCUSSION THREAD

**Discussion by kgpayne** [meltano/sdk#1249] [General]
As the number of Taps and Targets published to PyPI grows, name clashes become inevitable (i.e. two variants of `tap-dynamodb` - of which there are already 10 variants on the [hub](https://hub.meltano.com/extractors/tap-dynamodb/)). Therefore it would be useful to discuss ways to mitigate this by 'namespacing' Taps and Targets names in PyPI in a consistent way.

One common approach is to prepend a maintainer to the Tap/Target. E.g. the team at [Wise](https://pypi.org/user/analytics_platform/) prepend `pipelinewise-` to their Taps/Target names. E.g. `pipelinewise-tap-snowflake`. Crucially, this convention only applies to package distribution; the Tap is still executed using `tap-snowflake`.

**Comment by aaronsteers** [comment]
A proposal that we are coming around to internally would be to use `meltano-` or `meltanolabs-` as the prefix on the PyPi repo:

```
meltanolabs-tap-github
meltanolabs-tap-snowflake
meltanolabs-target-snowflake
```

Or:

```
meltano-tap-github
meltano-tap-snowflake
meltano-target-snowflake
```
